lib/opal/nodes/call.rb

```diff
@@ -261,38 +248,64 @@ def compile_default?
           first_arg = first_arg.updated(nil, [full_path])
         end
         @arglist = arglist.updated(nil, [first_arg] + rest)
-        compile_default!
-        push fragment('')
+        compile_default.call
       end
 
-      add_special :block_given? do
+      add_special :block_given? do |compile_default|
         push compiler.handle_block_given_call @sexp
       end
 
-      add_special :__callee__ do
+      add_special :__callee__ do |compile_default|
         if scope.def?
           push fragment scope.mid.to_s.inspect
         else
           push fragment 'nil'
         end
       end
 
-      add_special :__method__ do
+      add_special :__method__ do |compile_default|
         if scope.def?
           push fragment scope.mid.to_s.inspect
         else
           push fragment 'nil'
         end
       end
 
-      add_special :debugger do
+      add_special :debugger do |compile_default|
         push fragment 'debugger'
       end
 
-      add_special :__OPAL_COMPILER_CONFIG__ do
+      add_special :__OPAL_COMPILER_CONFIG__ do |compile_default|
         push fragment "Opal.hash({ arity_check: #{compiler.arity_check?} })"
       end
 
+      add_special :nesting do |compile_default|
+        recv, method, *args = children
+        push_nesting = push_nesting?(children)
+        push '(Opal.Module.$$nesting = $nesting, ' if push_nesting
+        compile_default.call
+        push ')' if push_nesting
+      end
+
+      add_special :constants do |compile_default|
+        recv, method, *args = children
+        push_nesting = push_nesting?(children)
+        push '(Opal.Module.$$nesting = $nesting, ' if push_nesting
+        compile_default.call
+        push ')' if push_nesting
+      end
+
+      def push_nesting?(recv)
+        recv = children.first
+
+        children.size == 2 && (           # only receiver and method
+          recv.nil? || (                  # and no receiver
```

Comment:
I think it could be replaced with `recv.nil? || recv == s(:const, nil, :Module)`. Is it written this way to cover both `Module` and `::Module` (with `nil` and `s(:cbase)` as constant scopes)?